install_base_ubuntu.sh

```diff
@@ -42,13 +42,22 @@ install_sdk() {
     fi
 }
 
+install_nodejs() {
+    # Dowload and install v8 of NodeJS, required for ipywidgets
+    curl -sL https://deb.nodesource.com/setup_8.x | bash -
```

Comment:
Maybe we should ask the user first before adding package sources, as it is a quite invasive and security sensitive operation (that repo could be used to replace any package in the system by a malicious one).

A --yes option would bypass that for automated setups (like travis or in Vagrantfile).

Reply:
There is already  an "--install-nodejs" option... if the user pass it then that should be fine.
Otherwise we will end up with two options: one to request to install nodejs and nother to say I don't care about my sources... that option is really meant for automated installations within a VM... or for trustful/lazy users (like I am).

Reply:
As a user, I would never expect a script to try to install something else outside of my distribution repository that can have a system-wide effect. Adding a new repository clearly have such system-wide effect with security implications, that's why I would prefer to treat it in a slightly different way. If the nodejs repo is hijacked, you could end up with a malicious kernel installed at the next update, and trusting a third-party repository or not is a decision of the user. That nodejs repo could also break existing other programs by distributing versions of libraries that are not the same as the one shipped in the distro repositories.

> that option is really meant for automated installations within a VM

We advertise running install_base_ubuntu.sh as root as part of the normal installation procedure:
https://lisa-linux-integrated-system-analysis.readthedocs.io/en/next/overview.html#installation

If we want to keep that script for VM setup only, we should have the list of ubuntu packages in a separate file and ask users to install that, rather than running that script.